### PR TITLE
Implemented Vert.x simple client support

### PR DESCRIPTION
**Pull request by pmlopes**
This is a simple implementation of a Vert.x Web handler for the the simple client

@brian-brazil and @beorn7 I've followed the servlet code in order to implement the Vert.x code, please let me know if this is the correct way.

Thanks!

### CODE REVIEW

File: simpleclient_vertx/pom.xml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>1.8</source>
```
Comment (brian-brazil) [inline simpleclient_vertx/pom.xml]: We support Java 6

File: simpleclient_vertx/src/main/java/io/prometheus/client/vertx/MetricsHandler.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
+package io.prometheus.client.vertx;
+
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.exporter.common.TextFormat;
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.RoutingContext;
+
+import java.io.*;
```
Comment (brian-brazil) [inline simpleclient_vertx/src/main/java/io/prometheus/client/vertx/MetricsHandler.java]: All imports should be listed.

File: simpleclient_vertx/pom.xml (hunk 1 above)
Comment (pmlopes) [inline simpleclient_vertx/pom.xml]: I'm fine with lowering it to 6, however vert.x requires 8 to run. In this case I can use abstract inner classes. What would you recommend?

Comment (brian-brazil) [inline simpleclient_vertx/pom.xml]: If it only works on 8 then that's okay, but avoid using more 8 syntax than you have to avoid confusion.

**Comment by pmlopes** [comment]
The only java 8 language feature in the code is a call to a default static method on an interface, this interfaces are part of vert.x core and it is the factory for buffers, and on the example the factory for vertx itself and the web router.

```java
Vertx.vertx()
Buffer.buffer()
Router.create(vertx)
```

I can't really avoid them. The only i can avoid is the method reference on the example which i can wrap in a annonymous inner class.

File: simpleclient_vertx/src/main/java/io/prometheus/client/vertx/MetricsHandler.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+   */
+  public MetricsHandler(CollectorRegistry registry) {
+    this.registry = registry;
+  }
+
+  @Override
+  public void handle(RoutingContext ctx) {
+
+    try {
+
```
Comment (brian-brazil) [inline simpleclient_vertx/src/main/java/io/prometheus/client/vertx/MetricsHandler.java]: You can trim the vertical whitespace a bit

**Comment by brian-brazil** [comment]
Can you squash your commits please?

**Comment by pmlopes** [comment]
Squashed!

**Comment by brian-brazil** [comment]
The CI isn't liking your use of java 8, you'll need to tweak travis.yml to use jdk8 for this module.

File: .travis.yml [hunk 4]
```diff
@@ -1,5 +1,8 @@
 language: java
 
+jdk:
+  - oraclejdk8
```
Comment (brian-brazil) [inline .travis.yml]: Can we still make sure the rest works with 6/7? I want to avoid an accidental regression here, as 2/3 of our users still are on 6.

Comment (pmlopes) [inline .travis.yml]: The top level pom forces 1.6:

https://github.com/prometheus/client_java/blob/master/pom.xml#L63

and only the vert.x one allows 1.8

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**